File: Simulation/0999.Available Captures for Rook/Solution.java
class Solution {
    public int numRookCaptures(char[][] board) {
        int m = board.length, n = board[0].length;
        int x = 0, y = 0;
        for (int i = 0; i < m; i++) {
            for (int j = 0; j < n; j++) {
                if (board[i][j] == 'R') {
                    x = i;
                    y = j;
                    break;
                }
            }
        }
        int ans = 0;
        for (int[] d : new int[][]{{1, 0}, {0, 1}, {-1, 0}, {0, -1}}) {
            int nx = x + d[0], ny = y + d[1];
            while (0 <= nx && nx < m && 0 <= ny && ny < n) {
                if (board[nx][ny] != '.') {
                    if (board[nx][ny] == 'p') {
                        ans++;
                    }
                    break;
                }
                nx += d[0];
                ny += d[1];
            }
        }
        return ans;
    }
}

File: Simulation/0999.Available Captures for Rook/Solution.py
class Solution:
    def numRookCaptures(self, board: List[List[str]]) -> int:
        m, n = len(board), len(board[0])
        x = y = 0
        for i in range(m):
            for j in range(n):
                if board[i][j] == 'R':
                    x, y = i, j
                    break
        ans = 0
        for d in [[1, 0], [0, 1], [-1, 0], [0, -1]]:
            nx, ny = x + d[0], y + d[1]
            while 0 <= nx < m and 0 <= ny < n:
                if board[nx][ny] != '.':
                    if board[nx][ny] == 'p':
                        ans += 1
                    break
                nx += d[0]
                ny += d[1]
        return ans